Description: Adapt the configure script to FreeBSD.
 - use the elf.h header;
 - shorten the library .so file name;
 - honor LDFLAGS when linking;
 - do not link against or depend on libc.
Forwarded: not-needed
Author: Kevin Van Maren <vanmaren@fast.cs.utah.edu>,
	Maxim Sobolev <sobomax@FreeBSD.org>,
	Peter Pentchev <roam@FreeBSD.org>
	Ying-Chieh Liao <ijliao@FreeBSD.org>
Last-Update: 2009-11-20

--- configure.orig	2014-01-26 00:46:23 UTC
+++ configure
@@ -734,7 +734,7 @@ cat > conftest.$ac_ext << EOF
 #line 735 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:740: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
@@ -1114,6 +1114,7 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 1116 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <ctype.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -1146,7 +1147,7 @@ EOF
 
 fi
 
-for ac_hdr in unistd.h stdint.h fcntl.h
+for ac_hdr in unistd.h stdint.h fcntl.h elf.h
 do
 ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
 echo $ac_n "checking for $ac_hdr""... $ac_c" 1>&6
@@ -1483,8 +1484,9 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 1485 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
-main()
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1522,8 +1524,9 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 1524 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
-main()
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1561,8 +1564,9 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 1563 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
-main()
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1600,8 +1604,9 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 1602 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
-main()
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -1640,8 +1645,9 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 1642 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
-main()
+int main()
 {
   FILE *f=fopen("conftestval", "w");
   if (!f) exit(1);
@@ -2258,6 +2264,7 @@ else
    The main things grep needs to know about mmap are:
    * does it exist and is it safe to write into the mmap'd area
    * how to use it (BSD variants)  */
+#include <stdlib.h>
 #include <sys/types.h>
 #include <fcntl.h>
 #include <sys/mman.h>
@@ -3281,7 +3288,7 @@ else
 fi
 
     case "$host" in
-      *-linux*|*-gnu*)
+      *-linux*|*-gnu*|*-dragonfly*|*-freebsd*|*-netbsd*|*-solaris*|*-midnight*)
 	if test "$GCC" = yes; then
 	  
   
@@ -3296,6 +3303,7 @@ else
   cat > conftest.$ac_ext <<EOF
 #line 3298 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 int
 main(int argc, char **argv) {
@@ -3336,13 +3344,12 @@ echo "$ac_t""$mr_cv_target_elf" 1>&6
 	    PICFLAGS='-fPIC -DPIC'
 	    if test "$mr_enable_gnu_names" = yes
 	    then SHLIB_SFX='-$(VERSION).so'
-	    else SHLIB_SFX='.so.$(VERSION)'
+	    else SHLIB_SFX='.so.$(MAJOR)'
 	    fi
 	    SHLINK_SFX='.so'
 	    SONAME_SFX='.so.$(MAJOR)'
-	    LINK_SHLIB='$(CC) -shared -Wl,-soname,$(SONAME)'
+	    LINK_SHLIB='$(CC) -shared -Wl,-soname,$(SONAME) $(LDFLAGS)'
 	    INSTALL_SHLIB='$(INSTALL_PROGRAM)'
-	    DEPSHLIBS='-lc'
 	  else
 	    echo "configure: warning: shared libraries not supported for $host" 1>&2
 	    mr_enable_shared=no
